libs/models/tournament.ts

```diff
@@ -0,0 +1,70 @@
+import { Player } from './player';
+import { Race } from './race';
+
+enum tournamentState {
+  /** **Lobby**
+   *
+   * Every players are in lobby.
+   * *(Race is not started yet.)*
+   *
+   */
+  lobby = 'lobby',
+
+  /** **Ready**
+   *
+   * Two or more players available in the lobby.
+   * *(Race is not started yet.)*
+   */
+  ready = 'ready',
+
+  /** **Full**
+   *
+   * The lobby is reached player limit.
+   * *(Race is not started yet.)*
+   */
+  full = 'full',
+
+  /** **Countdown**
+   *
+   * A player pressed the `Play` button.
+   */
+  countdown = 'countdown',
+
+  /** **Countdown**
+   *
+   * The race is ongoing.
+   */
+  race = 'race',
+
+  /** **Leaderboard**
+   *
+   * The race is finished.
+   */
+  leaderboard = 'leaderboard',
+
+  /** **Empty**
+   *
+   * Lobby has no players. Waiting to destroy the lobby.
+   */
+  empty = 'empty',
+}
+
+export interface Tournament {
+  /** Unique tournament id (/lobby ID) */
+  id: string;
+
+  /** Tournament state */
+  state: tournamentState;
+
+  /** Races history and currect race
+   *
+   * can be null if no races are started yet.
+   */
+  races: Race[] | null;
+
+  /** Players in the tournament
+   *
+   * can be null if lobby is empty.
+   */
+  players: Player[] | null;
```

Comment:
We can initialize as empty arrays right?
Removing the `null` state will help us avoid extra null check conditions in the code base

Reply:
Changed